### PR TITLE
Sscs 8774 select audio video evidence

### PR DESCRIPTION
**Pull request by jhart-dev**
added multiple evidence functionality for process audio video evidence

Jira Ticket: [https://tools.hmcts.net/jira/browse/SSCS-8774](url)

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/processaudiovideo/ProcessAudioVideoEvidenceAboutToSubmitHandler.java [hunk 1]
```diff
@@ -103,27 +107,30 @@ private void processIfIncludeEvidence(SscsCaseData caseData, PreSubmitCallbackRe
             List<DwpDocument> dwpDocuments = new ArrayList<>();
 
             for (AudioVideoEvidence audioVideoEvidence : caseData.getAudioVideoEvidence()) {
+                if (isSelectedEvidence(audioVideoEvidence, caseData)) {
```
Comment (rasikasri) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/processaudiovideo/ProcessAudioVideoEvidenceAboutToSubmitHandler.java:110]: I think you can make this logic simple by fetching the selected file first and then apply the logic to add it into the right collection. You don't need to do that inside the loop as only one file is selected.

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/processaudiovideo/ProcessAudioVideoEvidenceAboutToSubmitHandler.java [hunk 2]
```diff
@@ -103,27 +107,30 @@ private void processIfIncludeEvidence(SscsCaseData caseData, PreSubmitCallbackRe
             List<DwpDocument> dwpDocuments = new ArrayList<>();
 
             for (AudioVideoEvidence audioVideoEvidence : caseData.getAudioVideoEvidence()) {
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/processaudiovideo/ProcessAudioVideoEvidenceAboutToSubmitHandler.java:109]: Do you need to iterate through the list now? If you know which one was selected through the 'selectedAudioVideoEvidence' object, can't you just use that?

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/processaudiovideo/ProcessAudioVideoEvidenceAboutToSubmitHandler.java [hunk 3]
```diff
@@ -103,27 +107,30 @@ private void processIfIncludeEvidence(SscsCaseData caseData, PreSubmitCallbackRe
...
+                if (isSelectedEvidence(audioVideoEvidence, caseData)) {
 
-                if (UploadParty.DWP.equals(audioVideoEvidence.getValue().getPartyUploaded())) {
-                    dwpDocuments.add(buildAudioVideoDwpDocument(audioVideoEvidence, response));
-                } else {
-                    sscsDocuments.add(buildAudioVideoSscsDocument(audioVideoEvidence, response));
-                }
-            }
+                    if (UploadParty.DWP.equals(audioVideoEvidence.getValue().getPartyUploaded())) {
+                        dwpDocuments.add(buildAudioVideoDwpDocument(audioVideoEvidence, response));
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/processaudiovideo/ProcessAudioVideoEvidenceAboutToSubmitHandler.java:113]: Remove the buildAudioVideoDwpDocument method. You won't need to build it anymore as you are already building it as part of the midevent callback.

Comment (rasikasri) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/processaudiovideo/ProcessAudioVideoEvidenceAboutToSubmitHandler.java:113]: also you can use DocumentUtil util class for file extention check, there are couple of methods already.

**Review by jackmaloney — APPROVED** [review]

**Pull request merged**